ewatercycle/models/pcrglobwb.py

```diff
@@ -15,94 +17,175 @@
 from ewatercycle.parametersetdb.config import CaseConfigParser
 
 
+@dataclass
+class PCRGlobWBParameterSet:
+    """Parameter set for the PCRGlobWB model class.
+
+    A valid pcrglobwb parameter set consists of a folder with input data files
+    and should always include a default configuration file.
+    """
+    input_dir: Union[str, PathLike]
+    """Input folder path."""
+    default_config: Union[str, PathLike]
+    """Path to (default) model configuration file consistent with `input_data`."""
+    def __setattr__(self, name: str, value: Union[str, PathLike]):
+        self.__dict__[name] = Path(value).expanduser().resolve()
+
+
+@dataclass
+class PCRGlobWBForcing:
+    """Forcing data for the PCRGlobWB model class."""
+    precipitationNC: Optional[Union[str, PathLike]] = None
+    """Input file for precipitation data."""
+    temperatureNC: Optional[Union[str, PathLike]] = None
+    """Input file for temperature data."""
+    refETPotFileNC: Optional[Union[str, PathLike]] = None
+    """Input file for reference potential evapotranspiration data."""
+    def __setattr__(self, name: str, value: Union[str, PathLike, None]):
+        if value is not None and Path(value).exists():
+            self.__dict__[name] = Path(value).expanduser().resolve()
+        else:
+            warnings.warn(
+                f"No valid input path received for {name}, defaulting to ''.")
+            self.__dict__[name] = ''
+
+
 class PCRGlobWB(AbstractModel):
     """eWaterCycle implementation of PCRGlobWB hydrological model.
 
-    Attributes
+    Args:
+
+        version: pick a version from :py:attr:`~available_versions`
+        parameter_set: instance of :py:class:`~PCRGlobWBParameterSet`.
```

Comment:
forcing argument is undocumented
```suggestion
        parameter_set: instance of :py:class:`~PCRGlobWBParameterSet`.
        forcing: a forcing data object. If not given uses forcing from parameter_set.
```